### ISSUE TITLE
Middlewares support ?

### ISSUE THREAD

**Issue by wuuer** [dxx/feignhttp#12]
Something like [reqwest-middleware](https://github.com/TrueLayer/reqwest-middleware)

**Comment by holmofy** [comment]
Yes, it would be better if it could support retry, tracing and other functions

**Comment by dxx** [comment]
Yes, that's a great idea. I have time to think about how to do it.

**Comment by dxx** [comment]
Hi, guys, the new version [0.6.0-rc.1](https://github.com/dxx/feignhttp/releases/tag/v0.6.0-rc.1) has been released, supports custom client add support for reqwest middleware.

Reference examples:
- [custom client](https://github.com/dxx/feignhttp/blob/dev/examples/custom_client.rs)
- [reqwest middleware](https://github.com/dxx/feignhttp/blob/dev/examples/reqwest_middleware.rs)
